### PR TITLE
Fix/cdata throttling on ajax requests

### PR DESCRIPTION
**Pull request by circlecube**
## Proposed changes

Since we are unable to acquire AccessToken on ajax requests, bail in the connect method when we're doing ajax. This will still return the data if it's available, but will not attempt to connect when ajaxing.

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: includes/Customer.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Customer {
 	/**
 	 * Collect customer data
 	 *
-	 * @return array of customer data
+	 * @return array of customer data or false if none
```
Comment (wpscholar) [inline includes/Customer.php]: @circlecube We should probably coerce `false` to be an empty array and keep a consistent return type.

**Review by wpscholar — APPROVED** [review]

File: includes/Customer.php (hunk 1 above)
Comment (circlecube) [reply includes/Customer.php]: Does this look better, @wpscholar ?

File: includes/Customer.php [hunk 2]
```diff
@@ -248,6 +256,29 @@ public static function connect( $path ) {
...
+			return false;
+		}
+
+		// Does user have permissions or is this cron?
+		if ( ! ( current_user_can( 'manage_options' ) || wp_doing_cron() ) ) {
+			// Unable to acquire AccessToken either without user priveledges or for cron events
+			return false;
+		}
+
+		return true;
```
Comment (earnjam) [inline includes/Customer.php]: What about just something like this: 

```suggestion
		return ( current_user_can( 'manage_options' ) || wp_doing_cron() );
```

Comment (circlecube) [reply includes/Customer.php]: I like this, and am happy to bring it in

**Review by wpscholar — APPROVED** [review]

**Pull request merged**